client_test.go

```diff
@@ -38,6 +38,53 @@ func TestGetMarathonURL(t *testing.T) {
 	assert.Equal(t, endpoint.Client.GetMarathonURL(), endpoint.URL)
 }
 
+func TestApiRequest(t *testing.T) {
+	cases := []struct {
+		Auth     bool
+		Username string
+		Password string
+		Ok       bool
+	}{
+		{
+			Username: "bad",
```

Comment:
Does it make sense to set `Username` if `Auth` is false at the same time? The test seems to ignore credentials if we disable `Auth`.